lib/LedgerSMB/X12/EDI850.pm

```diff
@@ -66,25 +66,7 @@ sub _order {
                 $sender_idx = $elements[5];
                 $sender_id = $elements[6];
                 $form->{edi_isa} = \@elements;
-                my @new_elements = (
-                   $elements[0],
-                   $elements[1],
-                   $elements[2],
-                   $elements[3],
-                   $elements[4],
-                   $elements[7],
-                   $elements[8],
-                   $elements[5],
-                   $elements[6],
-                   $elements[9],
-                   $elements[10],
-                   $elements[11],
-                   $elements[12],
-                   $elements[13],
-                   $elements[14],
-                   $elements[15],
-                   $elements[16],
-                );
+                my @new_elements = @elements[0 ..4, 7, 8, 5, 6, 9 .. 16 ];
```

Comment:
Though not authoritative, page 3 of http://www.1edisource.com/Portals/0/docs/850V5010.pdf seems to refer to the proper ANSI specification.  Our indices clearly refer to the header elements sequenced by the ISO prefixes.
Using the descriptions on that page to generate constant names seems somewhat helpful.